### PR TITLE
Fix/simplify mqtt

### PR DESCRIPTION
**Pull request by cecinestpasunepipe**
Simplify MQTT loader to use Loader.find_all().

### CODE REVIEW

**Comment by Schamper** [comment]
Can you add a unit test that yields multiple targets? Maybe after https://github.com/fox-it/dissect.target/pull/574 is merged in?

File: dissect/target/loaders/mqtt.py [hunk 1]
```diff
@@ -252,20 +250,18 @@ def connect(self) -> None:
...
     def __init__(self, path: Union[Path, str], **kwargs):
         super().__init__(path)
         cls = MQTTLoader
+        self.broker = cls.broker
+        self.connection = MQTTConnection(self.broker, path)
 
-        if str(path).startswith("/remote/hosts/host"):
-            self.path = path.read_text()  # update path to reflect the resolved host
-
+    def find_all(path: Path, **kwargs):
```
Comment (Schamper) [inline dissect/target/loaders/mqtt.py:263]: Can you add a return type hint?

File: dissect/target/loaders/mqtt.py [hunk 2]
```diff
@@ -275,25 +271,15 @@ def __init__(self, path: Union[Path, str], **kwargs):
...
-            vfs.map_file_fh(self.PATH, BytesIO("\n".join(self.peers).encode("utf-8")))
-            for index, peer in enumerate(self.peers):
-                vfs.map_file_fh(f"{self.FOLDER}/host{index}-{peer}", BytesIO(peer.encode("utf-8")))
-
-            target.fs.mount("/data", vfs)
-            target.filesystems.add(vfs)
+        for disk in self.connection.info():
+            target.disks.add(RawContainer(disk))
 
     @staticmethod
```
Comment (Schamper) [inline dissect/target/loaders/mqtt.py:283]: Can you put the methods in the order `__init__`, `detect`, `find_all`, `map`? I know it doesn't matter, but it's more consistent with the rest of the loaders.

File: dissect/target/plugins/child/mqtt.py
Comment (Schamper) [inline dissect/target/plugins/child/mqtt.py:1]: Can't this file be removed? I remember it being removed in an earlier version of this PR?

**Review by Schamper — CHANGES_REQUESTED** [review]
Also seems something went wrong rebasing.

File: dissect/target/loaders/mqtt.py [hunk 3]
```diff
@@ -275,26 +275,11 @@ def __init__(self, path: Union[Path, str], **kwargs):
             cls.broker.connect()
             num_peers = int(options.get("peers", 1))
 
-        self.broker = cls.broker
-        self.connection = MQTTConnection(self.broker, self.path)
-        self.peers = self.connection.topo(num_peers)
+        cls.connection = MQTTConnection(cls.broker, path)
+        cls.peers = cls.connection.topo(num_peers)
+        for index, peer in enumerate(cls.peers):
+            yield peer
```
Comment (Schamper) [inline dissect/target/loaders/mqtt.py:281]:
```suggestion
        yield from cls.peers
```

File: tests/loaders/test_mqtt.py [hunk 4]
```diff
@@ -61,20 +63,21 @@ def mock_client(mock_paho: MagicMock) -> Iterator[MagicMock]:
...
+        ("group3", ["host4", "host5"], [10, 10, 1], 1, 1, 3, b"\x01\x02\x03"),  # + use disk 2
     ],
 )
 @patch.object(time, "sleep")  # improve speed during test, no need to wait for peers
 def test_remote_loader_stream(
     time: MagicMock,
     mock_client: MagicMock,
     alias: str,
-    host: str,
+    hosts: list[str],
```
Comment (Schamper) [inline tests/loaders/test_mqtt.py:80]: The other `hosts` parameters are still just strings?

**Review by Schamper — CHANGES_REQUESTED** [review]

File: tests/loaders/test_mqtt.py (hunk 4 above)
Comment (cecinestpasunepipe) [reply tests/loaders/test_mqtt.py:80]: Good find, funny that the test still worked, but it makes sense.

**Review by Schamper — APPROVED** [review]
Can you change the commit message/PR title to something more descriptive?

**Pull request merged**